relay-event-normalization/src/event.rs

```diff
@@ -753,80 +754,117 @@ fn normalize_performance_score(
             if !condition.matches(event) {
                 continue;
             }
-            if let Some(measurements) = event.measurements.value_mut() {
-                let mut should_add_total = false;
-                if profile.score_components.iter().any(|c| {
-                    !measurements.contains_key(c.measurement.as_str())
-                        && c.weight.abs() >= f64::EPSILON
-                        && !c.optional
-                }) {
-                    // All non-optional measurements with a profile weight greater than 0 are
-                    // required to exist on the event. Skip calculating performance scores if
-                    // a measurement with weight is missing.
-                    break;
-                }
-                let mut score_total = 0.0;
-                let mut weight_total = 0.0;
-                for component in &profile.score_components {
-                    // Skip optional components if they are not present on the event.
-                    if component.optional
-                        && !measurements.contains_key(component.measurement.as_str())
-                    {
+            let Some(measurements) = event.measurements.value_mut() else {
+                return;
+            };
+            normalize_performance_score_inner(measurements, profile);
```

Comment:
Once we found a matching profile, we can stop.

```suggestion
            normalize_performance_score_inner(measurements, profile);
            return;
```